internal/metrics/metrics_test.go

```diff
@@ -138,20 +132,16 @@ func TestFetchMirrorAvailabilityMetric500StatusCode(t *testing.T) {
 }
 
 func TestUpdateMetrics(t *testing.T) {
-	m, cfg, reg := setup()
+	m, cfg := setup()
 
 	timestamp := time.Date(2006, time.January, 2, 15, 4, 5, 0, time.UTC)
 	ts := createTestServer(timestamp, http.StatusOK)
 	defer ts.Close()
 
-	pushGateway := createTestPushGateway()
-	defer pushGateway.Close()
-
 	cfg.MirrorAvailabilityUrl = ts.URL
 	cfg.MetricRefreshInterval = 1 * time.Second
-	cfg.PushGatewayUrl = pushGateway.URL
 
-	go UpdateAndPushMirrorResponseStatusCode(m, cfg, reg)
+	go UpdateMirrorResponseStatusCode(m, cfg)
 
 	time.Sleep(2 * time.Second)
```

Comment:
Instead of sleeping, can these use the `synctest` package @nimalank7 used elsewhere in this codebase?

Reply:
ok, I'll try that out